sphinx/writers/latex.py

```diff
@@ -275,6 +275,23 @@ def escape_abbr(text):
     return re.sub('\.(?=\s|$)', '.\\@', text)
 
 
+def rstdim_to_latexdim(width_str):
+    """Convert `width_str` with rst length to LaTeX length."""
+    match = re.match('^(\d*\.?\d*)\s*(\S*)$', width_str)
+    if not match:
+        raise ValueError
+    res = width_str
+    amount, unit = match.groups()[:2]
+    float(amount)  # validate amount is float
```

Comment:
The regexp above is not strict to support multiple representations of amount. `1`, `1.0`, `1.` and `.5` are allowed. It matches even empty string or only dot (`.`).
This ensures the amount is valid float.

I know we can use more strict regexp instead. But it is a bit complicated and not readable.